cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/AttributeMultipleValuesInput/index.tsx

```diff
@@ -219,38 +223,30 @@ const AttributeMultipleValuesInputView = ({
     setCurrentInput(getCurrentWidgetAttributeValues(newType));
 
     // When user switches the 'selectedType', the value will reset back to an empty array.
-    setWidgetToAttributes((prevObjs) => ({
-      ...prevObjs,
-      [widgetID]: { ...prevObjs[widgetID], [field]: [] },
-    }));
+    setWidgetToAttributes(fromJS(widgetToAttributes).setIn([widgetID, field], fromJS([])));
   };
 
   const onAttributeChange = (newVal) => {
     setCurrentInput(newVal);
-    setWidgetToAttributes((prevObjs) => ({
-      ...prevObjs,
-      [widgetID]: { ...prevObjs[widgetID], [field]: newVal.split(COMMON_DELIMITER) },
-    }));
+    setWidgetToAttributes(
+      fromJS(widgetToAttributes).setIn([widgetID, field], fromJS(newVal.split(COMMON_DELIMITER)))
+    );
   };
 
   const onKeyValueAttributeChange = (keyvalue, type: SupportedType) => {
+    let keyvaluePairs;
     if (type === SupportedType.ValueLabelPair) {
-      const keyvaluePairs = keyvalue.pairs.map((pair) => {
+      keyvaluePairs = keyvalue.pairs.map((pair) => {
```

Comment:
Do we need the if/else here? It seems like keyvaluePairs is the same no matter what

Reply:
Resolved